netlogo-gui/src/main/lab/gui/ProgressDialog.scala

```diff
@@ -139,6 +136,33 @@ private [gui] class ProgressDialog(dialog: java.awt.Dialog, supervisor: Supervis
       workspace.jobManager.finishSecondaryJobs(null)
     }
   }
+
+  def updateView(check: Boolean): Unit = {
+    displaySwitch.setSelected(check)
+    workspace.displaySwitchOn(check)
+  }
+
+  def setUpdateView(status: Boolean): Unit = {
+    updateView(status)
+  }
+
+  def plotsAndMonitorsSwitch(check: Boolean): Unit = {
+    if(check){
+      plotsAndMonitorsSwitch.setSelected(true)
+      workspace.setPeriodicUpdatesEnabled(true)
+    }
+    else {
+      plotsAndMonitorsSwitch.setSelected(false)
+      workspace.setPeriodicUpdatesEnabled(false)
```

Comment:
This method can be simplified like `setUpdateView` and `setPlotsAndMonitorsSwitch`.

Reply:
I already thought of doing this thing earlier, but problem is whenever value is false 
`workspace.jobManager.finishSecondaryJobs(null)``

Should be set to NULL